src/main/java/com/softserve/demo/service/impl/CustomerServiceImpl.java

```diff
@@ -72,7 +81,50 @@ public void addImageToCustomer(Integer id, String fileName) {
     }
 
     @Override
-    public CustomerDTO findCustomerByUserId(Integer id) {
-        return customerMapper.customerToCustomerDTO(customerRepository.findCustomerByUserId(id));
+    public CustomerDTO getCustomerByUserId(Integer id) {
+        return customerMapper.customerToCustomerDTO(customerRepository.getCustomerByUserId(id));
+    }
+
+    @Override
+    public Page<CustomerDTO> getCustomersByStatus(Pageable pageable, CustomerStatus status) {
+        return customerRepository.findByStatus(status,pageable).map(customerMapper::customerToCustomerDTO);
+    }
+
+    @Override
+    public CustomerDTO updateStatus(Integer id, String status) {
+        Customer customer = customerRepository.findById(id).orElseThrow(() -> new NotFoundException("Customer not found"));
+        customer.setStatus(CustomerStatus.valueOf(status));
+        customerRepository.save(customer);
+        return customerMapper.customerToCustomerDTO(customer);
+    }
+
+    @Override
+    public <T> Page<Customer> findAll(Specification<T> approved, int page, int numberOfProvidersOnPage, String sortBy) {
+        return customerRepository.findAll(approved,PageRequest.of(page, numberOfProvidersOnPage, Sort.by(sortBy).descending()));
+    }
+
+    @Override
+    public void addFavourite(Integer id, ProviderDTO providerDTO) {
+        Customer customer = customerRepository.findCustomerById(id);
+        List<Provider> list = customer.getFavourite();
+        list.add(providerMapper.providerDTOToProvider(providerDTO));
+        customer.setFavourite(list);
+        customerRepository.save(customer);
+    }
+
+    @Override
+    public void removeById(Integer customerId, Integer favouriteId) {
+        Customer customer = customerRepository.findCustomerById(customerId);
+        List<Provider> list = customer.getFavourite();
+        int numberOfFavouriteProvider = 0;
+        for (Provider provider : list) {
```

Comment:
list.removeIf(p -> p.id == favouriteId).
And i believe we do not need to `customer.setFavourite(list);` coz this list is already linked to customer